janusgraph-test/src/test/java/org/janusgraph/graphdb/tinkerpop/optimize/JanusGraphMultiQueryStepTest.java

```diff
@@ -0,0 +1,87 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.janusgraph.graphdb.tinkerpop.optimize;
+
+import org.apache.tinkerpop.gremlin.process.traversal.Traversal;
+import org.apache.tinkerpop.gremlin.process.traversal.traverser.util.TraverserSet;
+import org.janusgraph.graphdb.tinkerpop.optimize.step.JanusGraphMultiQueryStep;
+import org.janusgraph.graphdb.tinkerpop.optimize.step.MultiQueriable;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.params.provider.Arguments.arguments;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class JanusGraphMultiQueryStepTest {
+
+    @ParameterizedTest
+    @MethodSource("generateTestParameters")
+    public void testClone(Traversal.Admin traversal, boolean limitBatchSize, Collection<MultiQueriable> clients) {
+        JanusGraphMultiQueryStep originalStep = new JanusGraphMultiQueryStep(traversal, limitBatchSize);
+        clients.forEach(originalStep::attachClient);
+
+        JanusGraphMultiQueryStep clone = originalStep.clone();
+
+        assertEquals(limitBatchSize, clone.isLimitBatchSize());
+        assertEquals(originalStep.getClientSteps().size(), clone.getClientSteps().size());
+        assertTrue(clone.getClientSteps().containsAll(originalStep.getClientSteps()));
+        assertTrue(originalStep.getClientSteps().containsAll(clone.getClientSteps()));
+    }
+
+    @ParameterizedTest
+    @MethodSource("generateTestParameters")
+    public void testReset(Traversal.Admin traversal, boolean limitBatchSize, Collection<MultiQueriable> clients) {
+        JanusGraphMultiQueryStep originalStep = new JanusGraphMultiQueryStep(traversal, limitBatchSize);
+        clients.forEach(originalStep::attachClient);
+
+        originalStep.reset();
+
+        assertEquals(limitBatchSize, originalStep.isLimitBatchSize());
+        assertEquals(originalStep.getClientSteps().size(), clients.size());
+        assertTrue(clients.containsAll(originalStep.getClientSteps()));
+        assertTrue(originalStep.getClientSteps().containsAll(clients));
+    }
+
+    private static Stream<Arguments> generateTestParameters() {
```

Comment:
Codacy is wrong about this method being unused. It is used to generate the parameters for the two methods above.

Reply:
Yes, Codacy is wrong about it. We should just ignore that. It's hard for Codacy to make correct assumptions especially about the code which uses Reflection API.

Reply:
You could actually bypass this rule by changing the `private` modifier to `protected`.

Reply:
Just out of personal preference, I would rather follow best practices for the code than introducing changes to satisfy erroneous warnings.

Reply:
That's fine, never mind then. I thought codacy is enabled on master as well - turns out it's disabled.